SimPEG/electromagnetics/static/induced_polarization/simulation_2d.py

```diff
@@ -40,19 +36,18 @@ def fields(self, m):
             # re-uses the DC simulation's fields method
             self._f = super().fields(None)
 
-        if not self._dc_data_set:
+        if self._scale is None:
+            scale = Data(self.survey, np.full(self.survey.nD, self._sign))
+            f = self.fields_to_space(self._f)
             # loop through receievers to check if they need to set the _dc_voltage
             for src in self.survey.source_list:
                 for rx in src.receiver_list:
                     if (
                         rx.data_type == "apparent_chargeability"
-                        and rx._dc_voltage is None
+                        or self.data_type == "apparent_chargeability"
```

Comment:
again do we need this?

could be useful, if one wants to set the data type by passing the `data_type` to a simulation.

Reply:
Again, I think it should be more noted as deprecated behavior.